atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -78,8 +81,20 @@ public CassandraClientPoolingContainer(
         this.config = config;
         this.poolNumber = poolNumber;
         this.poolMetrics = poolMetrics;
-        this.clientPool = createClientPool();
+        Clock clock = Clock.systemUTC();
+        TimingOutEvictionPolicy<CassandraClient> evictionPolicy = new TimingOutEvictionPolicy<>(
+                new NonEvictionLoggingEvictionPolicy<>(new DefaultEvictionPolicy<>()), clock);
+        this.clientPool = createClientPool(evictionPolicy);
         this.timedRunner = TimedRunner.create(config.timeoutOnConnectionBorrow().toJavaDuration());
+
+        evictionBasedPoolClearer = () -> {
+            Instant now = clock.instant();
+            if (Duration.between(evictionPolicy.getLastEviction(), now).abs().getSeconds()
+                    > config.timeoutOnPoolEvictionFailure().toSeconds()) {
+                evictionPolicy.resetLastEviction();
+                clientPool.clear();
```

Comment:
If we want to roll this out more broadly, it seems like we should add a condition that the pool is full (or at least 90% full) - otherwise we create a lot of churn for stacks that are not super busy. I'm fine with testing in this configuration though.